test/lsp/cache_protocol_test_corpus.cc

```diff
@@ -258,4 +259,72 @@ TEST_CASE_FIXTURE(CacheProtocolTest, "LSPDoesNotUseCacheIfModified") {
         }
     }
 }
+
+TEST_CASE_FIXTURE(CacheProtocolTest, "ReindexingUsesTheCache") {
+    // Write a file to disk.
+    auto relativeFilepath = "test.rb";
+    auto filePath = fmt::format("{}/{}", rootPath, relativeFilepath);
+    // This file has an error to indirectly assert that LSP is actually typechecking the file during initialization.
+    auto fileContents = "# typed: true\n"
+                        "class Foo extend T::Sig\n"
```

Comment:
```suggestion
                        "class Foo\n"
                        "  extend T::Sig\n"
```

slight preference, not blocking

Reply:
unfortunately accepting this would make your `assertErrorDiagnostics` need to change too, so feel free to just ignore.

Reply:
That's what I get for trying to format the previous test case that I stole 😅

Reply:
thank you for at least trying, I have converted most of these but I guess I forgot about the `cache_*` test.